kubernetes/cache/kube_cache.go

```diff
@@ -2186,6 +2238,7 @@ func (c *kubeCache) GetRequestAuthentications(namespace, labelSelector string) (
 	for _, ra := range requestAuthentications {
 		raCopy := ra.DeepCopy()
 		raCopy.Kind = kubernetes.RequestAuthenticationsType
+		raCopy.APIVersion = kubernetes.ApiSecurityVersionV1
```

Comment:
as I remember not setting `APIVersion` was on purpose

Reply:
If the Istio object is created using `kubectl`, the `APIVersion` value is not added. The Istio config page doesn't actually need this value, but it is required for the OSSMC plugin. Maybe there is another way to figure out this value. Any idea @nrfox @hhovsepy?

Reply:
I'm not sure if it will solve this bug or not but I created: https://github.com/kiali/kiali/issues/7457 to track the `Kind` issue.

Reply:
So this is partially a kubernetes issue. See: https://github.com/kubernetes/kubernetes/issues/3030 for more details. And both `kind` and `apiVersion` are unset.

Reply:
Why is OSSMC using `apiVersion` but standalone is not?

Reply:
> Why is OSSMC using apiVersion but standalone is not?

The Istio config list page is the only native page of OSSMC to align with the rest of OpenShift list pages, so its code is different from that of Kiali standalone. The link to a specific Istio config item is completely different in both places, and OSSMC needs both the `kind` and `apiVersion` values.

**Kiali standalone:** `/namespaces/bookinfo/istio/virtualservices/bookinfo`

**OpenShift (OSMC):** `k8s/ns/bookinfo/networking.istio.io~v1~VirtualService/bookinfo`

In OSSMC, you need both `kind` and `apiVersion` values to create the Istio config item link, but this is not needed in Kiali standalone.

Actually, if this is a hack, I have no problem doing it in OSSMC as I already have this information  (https://github.com/kiali/openshift-servicemesh-plugin/blob/main/plugin/src/openshift/utils/IstioResources.ts). 

I created the issue because it seemed odd to me that you fulfill `kind` but not `apiVersion` when both are unset. In these cases, I usually expect to receive all related values or none.

Reply:
> Actually, if this is a hack, I have no problem doing it in OSSMC as I already have this information (https://github.com/kiali/openshift-servicemesh-plugin/blob/main/plugin/src/openshift/utils/IstioResources.ts).

Well it's a hack to get around a real kube client issue. I'm good with this change. When we address #7457 we can have a more generic solution since setting the typemeta manually is a bit error prone but there's probably some cases like gateway api where we actually want to set the wrong kind due to conflicts until #7457 is resolved.